### ISSUE TITLE
--strip=never is not respected in fastbuild

### ISSUE THREAD

**Issue by varungandhi-src** [uber/hermetic_cc_toolchain#31]
First of all, a quick thanks for creating an open-sourcing this project. I'm looking into building for Ubuntu 18.04 / Debian Buster (older glibc version), and using this toolchain seems simpler than other alternatives, especially given that GitHub Actions stopped providing Ubuntu 18.04 runners.

On to the actual issue. It looks like if you create a basic hello world project and use `bazel build //:main --strip=never`, then the stripping flag is not respected, there is a `-Wl,-S` at the end of the compiler invocation.

```
(cd /home/varun/.cache/bazel/_bazel_varun/938729434654b701ee0f9a4c2d2a514d/execroot/__main__ && \
  exec env - \
    BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1 \
    PATH=/home/varun/.cache/bazelisk/downloads/bazelbuild/bazel-6.1.1-linux-x86_64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin \
    PWD=/proc/self/cwd \
  external/zig_sdk/tools/x86_64-linux-musl/c++ -o bazel-out/k8-fastbuild/bin/main bazel-out/k8-fastbuild/bin/_objs/main/main.o -fno-lto -Wl,-S)
```

When using `-c opt` or `-c dbg`, this is not a problem.

Is this behavior intentional? It doesn't seem to match the default Bazel cc toolchain behavior.

**Comment by motiejus** [comment]
To be honest, this is the first time I heard about the `--strip` option and don't know how to change the linkopts following it.

Currently stripping is controlled here: https://github.com/uber/hermetic_cc_toolchain/blob/v2.0.0-rc1/toolchain/zig_toolchain.bzl#L79

I will accept a patch that would take into account `--strip` in that "fastbuild feature".

**Comment by motiejus** [comment]
Should be fixed by #48. Please confirm.

**Comment by motiejus** [comment]
### fastbuild

```
$ bazel run --ui_event_filters=-info,-stderr,-debug,-progress --noshow_progress --run_under=file //test/c:which_libc_linux_amd64_gnu.2.19
/home/motiejus/.cache/bazel/_bazel_motiejus/27579b5a6c0c322d194d6b503decd17d/execroot/hermetic_cc_toolchain/bazel-out/k8-fastbuild-ST-6aec337e4f98/bin/test/c/which_libc-__libc_aware_platform_linux_amd64_gnu.2.19: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.0.0, stripped
```

### fastbuild --strip=never

```
$ bazel run --ui_event_filters=-info,-stderr,-debug,-progress --noshow_progress --strip=never --run_under=file //test/c:which_libc_linux_amd64_gnu.2.19
/home/motiejus/.cache/bazel/_bazel_motiejus/27579b5a6c0c322d194d6b503decd17d/execroot/hermetic_cc_toolchain/bazel-out/k8-fastbuild-ST-6aec337e4f98/bin/test/c/which_libc-__libc_aware_platform_linux_amd64_gnu.2.19: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.0.0, with debug_info, not stripped
```

Thanks to  @jvolkman, this is now fixed.
